src/main/scala/ldbc/snb/datagen/factors/FactorGenerationStage.scala

```diff
@@ -42,6 +42,12 @@ object FactorGenerationStage extends DatagenStage with Logging {
       .select(by :+ $"frequency": _*)
       .orderBy($"frequency".desc +: by.map(_.asc): _*)
 
+  private def undirectedKnows(personKnowsPerson: DataFrame) =
+    personKnowsPerson.select($"Person1Id", $"Person2Id")
+      .union(personKnowsPerson.select($"Person2Id".as("Person1Id"), $"Person1Id".as("Person2Id")))
```

Comment:
I think it might be faster to do
```suggestion
    personKnowsPerson
        .select(expr("stack(2, Person1Id, Person2Id, Person2Id, Person1Id)").as(Seq("Person1Id", "Person2Id")))
```

Unfortunately `expr` is needed because the `stack` function is not exposed through the Scala API.
https://spark.apache.org/docs/latest/api/sql/index.html#stack
https://spark.apache.org/docs/3.2.0/api/java/org/apache/spark/sql/functions.html#expr-java.lang.String-